managedtenants/bundles/index_builder.py

```diff
@@ -88,3 +94,75 @@ def _push(self, index_image):
             err_msg = f"failed to push {index_image}: {e}."
             self.log.error(err_msg)
             raise IndexBuilderError(err_msg)
+
+    def _validate_sql_catalog(self, tag):
+        """
+        Validates an sql based index_image. Makes sure it contains bundles.
+        """
+        sql_catalog = None
+        try:
+            in_memory_db = self.docker_api.extract_file_from_container(
+                tag, "/database/index.db"
+            )
+
+            sql_catalog = SQLCatalog(in_memory_db)
+            bundles = sql_catalog.get_bundles()
+            if len(bundles) == 0:
+                err_msg = f"{tag} contains 0 bundles."
+                self.log.error(err_msg)
+                raise IndexBuilderError(err_msg)
+
+            self.log.debug(f"Found {len(bundles)} bundles:")
+            for bundle in bundles:
+                self.log.debug(bundle)
+
+        except DockerError as e:
+            err_msg = f"failed to validate {tag}, got {e}."
+            self.log.error(err_msg)
+            raise IndexBuilderError(err_msg)
+
+        except sqlite3.Error as e:
+            err_msg = (
+                "failed to query bundles from '/database/index.db' inside of ",
+                f"{tag}, got {e}.",
+            )
+            self.log.error(err_msg)
+            raise IndexBuilderError(err_msg)
+
+        finally:
+            if sql_catalog:
+                sql_catalog.close()
```

Comment:
Since this is a custom class I think adding `__enter__` and `__exit__` and using `SQLCatalog` as a context manager would keep things a little tidier.

Reply:
fixed